profileservice/api/controllers/profiles.py

```diff
@@ -234,6 +234,81 @@ def get_data_list(uuid):
 
     return None, None, True, resp
 
+def core_search(netid=None, firstname=None, lastname=None): 
+    if request.headers.get("ROKWIRE-CBB-API-KEY") != cfg.CORE_BB_API_KEY:
+        msg = {
+            "reason": "Unauthorized",
+            "error": "Unauthorized: " + request.url,
+        }
+        msg_json = jsonutils.create_log_json("CORE PROFILE", "GET", msg)
+        logging.error("CORE PROFILE GET " + json.dumps(msg_json))
+        return rs_handlers.unauthorized(msg_json) 
+
+    fields = {}
+    if netid:
+        fields['netid'] = netid
+    if firstname and lastname:
+        fields['firstname'] = firstname
+        fields['lastname'] = lastname
+    if len(fields) == 0:
+        msg = {
+            "reason": "Must provide firstname and lastname or netid",
+            "error": "Bad Request: " + request.url,
+        }
+        msg_json = jsonutils.create_log_json("CORE PROFILE", "GET", msg)
+        logging.error("CORE PROFILE GET " + json.dumps(msg_json))
+        return rs_handlers.bad_request(msg_json)
+
+    is_error = False
+    resp = None
+
+    if fields != None:
+        data_list = mongoutils.get_pii_result(fields)
+        if len(data_list) > 1:
+            msg = {
+                "reason": "There is more than 1 pii record: " + str(fields),
+                "error": "Not Found: " + request.url,
+            }
+            msg_json = jsonutils.create_log_json("CORE PROFILE", "GET", msg)
+            logging.error("CORE PROFILE GET " + json.dumps(msg_json))
+            is_error = True
+            resp = rs_handlers.not_found(msg_json)
+        if len(data_list) == 0:
+            msg = {
+                "reason": "There is no pii record for the query: " + str(fields),
+                "error": "Not Found: " + request.url,
+            }
+            msg_json = jsonutils.create_log_json("CORE PROFILE", "GET", msg)
+            logging.error("CORE PROFILE GET " + json.dumps(msg_json))
+            is_error = True
+            resp = rs_handlers.not_found(msg_json)
+    else:
+        msg = {
+            "reason": "Invalid search: " + str(fields),
+            "error": "Bad Request: " + request.url,
+        }
+        msg_json = jsonutils.create_log_json("CORE PROFILE", "GET", msg)
+        logging.error("CORE PROFILE GET " + json.dumps(msg_json))
+        is_error = True
+        resp = rs_handlers.bad_request(msg_json)
+
+    if is_error:
+        return resp
+
+    data_list = jsonutils.remove_file_descriptor_from_data_list(data_list)
+    uuid_list = data_list[0]['uuid']
```

Comment:
We need to check the length of this list to avoid invalid references and use `.get()` instead of `[]` to handle None values

Reply:
The length of ```data_list``` is checked in the lines above this snippet. ```.get()``` has been added.